### DISCUSSION TITLE
[Spotbugs Maven] Increasing the severity of an existing rule

### DISCUSSION THREAD

**Discussion by omindu** [spotbugs/spotbugs#1585] [Q&A] [answered]
Hi Devs,

Is there a mechanism to increase the severity of a particular rule? As of now we are executing the spotbugs plugin with `threshold` set to `High` but we want to include some of these  medium level rules in the high category.

For an example, `ODR - ODR_OPEN_DATABASE_RESOURCE` is currently getting reported when the threshold is set to `Medium` and we want to get this reported when the threshold is set to `High`.

Thanks!

**Comment by KengoTODA** [comment, marked as answer]
I haven't confirmed but it's theoretically possible.

Maven plugin's `threshold` config is [mapped to SpotBugs core's `priorityThreshold`](https://github.com/spotbugs/spotbugs-maven-plugin/blob/40ad0e93ec1488c674c9c27e5937effd31e84379/src/main/groovy/org/codehaus/mojo/spotbugs/SpotBugsMojo.groovy#L1169-L1201). For instance, `<threshold>high</threshold>` introduces a [`-high` option](https://spotbugs.readthedocs.io/en/stable/running.html#text-ui-options) to SpotBugs and [it configure reporter](https://github.com/spotbugs/spotbugs/blob/e2daa309052eb4230055b3947ec9212898a60f54/spotbugs/src/main/java/edu/umd/cs/findbugs/TextUICommandLine.java#L308-L309) to use.

This is a config of reporter, it means that,`ODR_OPEN_DATABASE_RESOURCE` bug instance itself is generated even when we use `<threshold>high</threshold>`.
So what we need is a way to keep `ODR_OPEN_DATABASE_RESOURCE` generated by `edu.umd.cs.findbugs.detect.FindOpenStream` detector reported to users. Then we have several ideas:

1. Stop using `<threshold>high</threshold>` and [use a filter file](https://spotbugs.github.io/spotbugs-maven-plugin/usage.html#Filter_bugs_to_report) instead. Filter provides [`<Confidence>` matcher](https://spotbugs.readthedocs.io/en/stable/filter.html#confidence) so you can configure reporters to report both `ODR_OPEN_DATABASE_RESOURCE` and other high priority bug instances.
2. SpotBugs core provides [`adjustPriority` property](https://spotbugs.readthedocs.io/en/stable/running.html#detector-visitor-configuration-options) to adjust priority of given visitor (bug detector). Please adjust priority of `edu.umd.cs.findbugs.detect.FindOpenStream` detector by this option. But I cannot find a way to configure `adjustPriority` property in [Maven plugin's Mojo](https://github.com/spotbugs/spotbugs-maven-plugin/blob/40ad0e93ec1488c674c9c27e5937effd31e84379/src/main/groovy/org/codehaus/mojo/spotbugs/SpotBugsMojo.groovy) so you probably need to fork it.

**Reply by omindu** [reply]
Thanks @KengoTODA for the details response! Being able to do this without forking it would be ideal.

**Reply by KengoTODA** [reply]
then please try the 1st way :)
